client/signup/controller.js

```diff
@@ -305,6 +306,16 @@ export default {
 		const isManageSiteFlow =
 			! excludeFromManageSiteFlows && ! isAddNewSiteFlow && isReEnteringSignupViaBrowserBack;
 
+		// Hydrate the store with domains dependencies from session storage,
+		// only in the onboarding flow.
+		const signupDependencies = getSignupDependencies();
+		if ( signupDependencies && isManageSiteFlow && flowName === 'onboarding' ) {
+			const { step, dependencies } = JSON.parse( signupDependencies );
+			if ( step && dependencies ) {
+				context.store.dispatch( submitSignupStep( step, dependencies ) );
```

Comment:
I don't think any additional security is needed here because the data is going through the usual submit checks and management. But let me know if you spot something.